shell/platform/darwin/ios/framework/Source/FlutterMetalLayerTest.mm

```diff
@@ -236,4 +236,42 @@ - (void)testLayerLimitsDrawableCount {
   [self removeMetalLayer:layer];
 }
 
+- (void)testTimeout {
```

Comment:
This needs more thought, I haven't actually looked at how to grab the usage flag in the test yet

Reply:
The test fails because `isInUse=true` doesn't prevent layer from returning the surface. 

`isInUse=false`: Compositor is definitely not using the surface. This is useful to reuse surface immediately after a dropped frame.
`isInUse=true`: Compositor may be using the surface. if there is no surface with `isInUse=false`, return one with `isUnUse=true` that was presented longest time ago.

The blocking in this PR is preformed when front surface command buffer has not been completed. This is different from compositor using the surface. The test will likely need to call `flutterPrepareForPresent:` with command buffer and then test if `nextSurface` blocks while the command buffer waits for completion.